### PR TITLE
Replace all fa-angle/fa-chevron icons with equivalent Icon components…

### PR DESCRIPTION
**Pull request by imdebamrita**
…. Issue No: #17448

I replaced all the fa-angle, fa-chevron, <IconCaret.. icons with equivalent Icon components…

### CODE REVIEW

File: ui/components/app/advanced-gas-inputs/advanced-gas-inputs.component.js [hunk 1]
```diff
@@ -177,7 +177,7 @@ export default class AdvancedGasInputs extends Component {
                 !disabled && onChange({ target: { value: value + 1 } })
```
Comment (georgewrmarshall) [inline ui/components/app/advanced-gas-inputs/advanced-gas-inputs.component.js:177]: You will have to import the `Icon` component and `ICON_NAMES` object

```
import { Icon, ICON_NAMES } from '../../component-library';
```

File: ui/components/app/asset-list-item/asset-list-item.js [hunk 2]
```diff
@@ -138,7 +138,7 @@ const AssetListItem = ({
       rightContent={
         !isERC721 && (
           <>
-            <i className="fas fa-chevron-right asset-list-item__chevron-right" />
+            <Icon name={ICON_NAMES.ARROW_RIGHT} />
```
Comment (georgewrmarshall) [inline ui/components/app/asset-list-item/asset-list-item.js:141]: Same here. It also might be worth checking what the additional className does

File: ui/components/app/confirm-page-container/confirm-page-container-header/confirm-page-container-header.component.js [hunk 3]
```diff
@@ -54,7 +54,7 @@ export default function ConfirmPageContainerHeader({
               visibility: showEdit ? 'initial' : 'hidden',
```
Comment (georgewrmarshall) [inline ui/components/app/confirm-page-container/confirm-page-container-header/confirm-page-container-header.component.js:54]: Same here

File: ui/components/app/confirm-page-container/confirm-page-container-navigation/confirm-page-container-navigation.component.js [hunk 4]
```diff
@@ -72,7 +72,7 @@ const ConfirmPageContainerNavigation = () => {
           data-testid="previous-page"
           onClick={() => onNextTx(prevTxId)}
         >
-          <i className="fa fa-angle-left fa-2x" />
```
Comment (georgewrmarshall) [inline ui/components/app/confirm-page-container/confirm-page-container-navigation/confirm-page-container-navigation.component.js:75]: You will have to import the `Icon` component and `ICON_NAMES` object and ensure the size matches the font awesome icons

```
import { Icon, ICON_NAMES } from '../../component-library';
```

File: ui/components/app/edit-gas-fee-button/edit-gas-fee-button.js [hunk 5]
```diff
@@ -71,7 +71,7 @@ export default function EditGasFeeButton({ userAcknowledgedGasMissing }) {
           </span>
         )}
         <span className="edit-gas-fee-button__label">{t(title)}</span>
-        <i className="fas fa-chevron-right asset-list-item__chevron-right" />
```
Comment (georgewrmarshall) [inline ui/components/app/edit-gas-fee-button/edit-gas-fee-button.js:74]: You will have to import the `Icon` component and `ICON_NAMES` object and check class name incase more styles are necesary

```
import { Icon, ICON_NAMES } from '../../component-library';
```

File: ui/components/app/modals/account-modal-container/account-modal-container.component.js [hunk 6]
```diff
@@ -24,7 +24,7 @@ export default function AccountModalContainer(props, context) {
         </div>
         {showBackButton && (
           <div className="account-modal__back" onClick={backButtonAction}>
-            <i className="fa fa-angle-left fa-lg" />
```
Comment (georgewrmarshall) [inline ui/components/app/modals/account-modal-container/account-modal-container.component.js:27]: You will have to import the `Icon` component and `ICON_NAMES` object

```
import { Icon, ICON_NAMES } from '../../component-library';
```

File: ui/components/ui/icon/icon.stories.js [hunk 7]
```diff
@@ -120,10 +120,10 @@ export const DefaultStory = (args) => (
...
         }}
       >
-        <IconItem Component={<IconCaretLeft {...args} />} />
-        <IconItem Component={<IconCaretRight {...args} />} />
-        <IconItem Component={<IconCaretDown {...args} />} />
-        <IconItem Component={<IconCaretUp {...args} />} />
+        <IconItem Component={<Icon name={ICON_NAMES.ARROW_LEFT} {...args} />} />
+        <IconItem Component={<Icon name={ICON_NAMES.ARROW_RIGHT} {...args} />} />
+        <IconItem Component={<Icon name={ICON_NAMES.ARROW_DOWN} {...args} />} />
+        <IconItem Component={<Icon name={ICON_NAMES.ARROW_UP} {...args} />} />
```
Comment (georgewrmarshall) [inline ui/components/ui/icon/icon.stories.js:126]: You could just remove these lines once IconCaret[dir] is no longer used

**Review by georgewrmarshall — COMMENTED** [review]
Hey @imdebamrita, thanks for your contribution! There is still a bit of work to do on this PR before we can merge it. You will have to import the `Icon` component and `ICON_NAMES` object to every file
```
import { Icon, ICON_NAMES } from '../../component-library';
```

As well as that it would be worth checking what styles the additional classNames do for the Icon component you may be able to use the utility props from the Box component that the Icon component accepts https://metamask.github.io/metamask-storybook/?path=/docs/components-ui-box--default-story#props

**Comment by imdebamrita** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by imdebamrita** [comment]
Thank you @georgewrmarshall sir.
I will fix those issues.

File: ui/components/app/asset-list-item/asset-list-item.js (hunk 2 above)
Comment (imdebamrita) [reply ui/components/app/asset-list-item/asset-list-item.js:141]: How do i check what those additional className does.....?

Comment (imdebamrita) [reply ui/components/app/asset-list-item/asset-list-item.js:141]: Also I don't know how I can run this extension in my local machine to check those things.......
Could you please help.

Comment (georgewrmarshall) [reply ui/components/app/asset-list-item/asset-list-item.js:141]: Hey @imdebamrita, having a local build is probably a prerequisite for taking on this task please read through our contributor docs https://github.com/MetaMask/metamask-extension#building-locally

**Comment by georgewrmarshall** [comment]
Hey @imdebamrita, it looks like this PR is out of date and since opening the work has been completed to fix this issue. Thanks for your contribution and I encourage you to contribute to the extension more in future.

**Pull request closed without merging**